Source/DafnyDriver/DafnyDoc.cs

```diff
@@ -0,0 +1,1074 @@
+    var imports = module.TopLevelDecls.Where(d => d is AliasModuleDecl).Select(d => d as AliasModuleDecl).ToList();
+    imports.Sort((f, ff) => f.Name.CompareTo(ff.Name));
+    var absimports = module.TopLevelDecls.Where(d => d is AbstractModuleDecl).Select(d => d as AbstractModuleDecl).ToList();
+    absimports.Sort((f, ff) => f.Name.CompareTo(ff.Name));
+    if (imports.Count() + absimports.Count() > 0) {
+      summaries.Append(Heading3("Imports"));
+      summaries.Append(TableStart());
+      foreach (var imp in imports) {
+        var info = ImportInfo(imp, register);
+        owner.Contents.Add(info);
+        summaries.Append(info.HtmlSummary).Append(eol);
+      }
+      foreach (var imp in absimports) {
+        var info = ImportInfo(imp, register);
+        owner.Contents.Add(info);
+        summaries.Append(info.HtmlSummary).Append(eol);
+      }
+      summaries.Append(TableEnd());
+    }
+  }
+
+  public void AddSubModuleSummaries(ModuleDefinition module, StringBuilder summaries, Info owner, bool register) {
+    var submods = module.TopLevelDecls.Where(d => d is LiteralModuleDecl).Select(d => d as LiteralModuleDecl).ToList();
+    submods.Sort((f, ff) => f.Name.CompareTo(ff.Name));
+    if (submods.Count() > 0) {
+      summaries.Append(Heading3("Submodules"));
+      summaries.Append(TableStart());
+      foreach (var submod in submods) {
+        var info = ModuleInfo(register, submod, submod.ModuleDef, null);
+        owner.Contents.Add(info);
+        summaries.Append(Row(info.HtmlSummary)).Append(eol);
+      }
+      summaries.Append(TableEnd());
+    }
+  }
+
+  public bool IsType(TopLevelDecl t) {
+    return (t is RevealableTypeDecl || t is SubsetTypeDecl);
+  }
+
+  public void AddTypeSummaries(ModuleDefinition module, StringBuilder summaries, Info owner, bool register) {
+    var types = module.TopLevelDecls.Where(c => IsType(c) && !IsGeneratedName(c.Name)).ToList();
+    types.Sort((f, ff) => f.Name.CompareTo(ff.Name));
+    if (types.Count() > 1 || (types.Count() == 1 && (types[0] is DefaultClassDecl))) {
+      summaries.Append(Heading3("Types"));
+      summaries.Append(TableStart());
+      foreach (var t in types) {
+        if (t is DefaultClassDecl) {
+          continue;
+        }
+        var info = TypeInfo(register, t, module, owner);
+        var styledName = Code(Bold(t.Name));
+        owner.Contents.Add(info);
+        summaries.Append(Row(info.HtmlSummary)).Append(eol);
+      }
+      summaries.Append(TableEnd());
+    }
+  }
+
+  public Info ConstantInfo(bool register, ConstantField c) {
+    var info = new Info(register, this, "const", c.Tok, c.Name, c.FullDafnyName);
+
+    var docstring = Docstring(c);
+    var modifiers = c.ModifiersAsString();
+    var linkedName = Code(Link(c.FullDafnyName, Bold(c.Name)));
+    var linkedType = TypeLink(c.Type);
+    var rhsstring = c.Rhs == null ? "" : (" := " + c.Rhs.ToString());
```

Comment:
```suggestion
    var rhsstring = c.Rhs == null ? "" : $" := {c.Rhs.ToString()}";
```
More readable than parentheses for me.